// bpmn-questionnaire/lib/components/Results.js
'use strict';

var h        = require('virtual-dom/h');

// Components
var Progress = require('./Progress.js'),
    Controls = require('./Controls.js');

/**
 * Results component.
 *
 * @constructor
 * @param {Object} questionnaire - Reference to questionnaire.
 */
function Results(questionnaire) {
  this.questionnaire = questionnaire;
}

/**
 * Rendering function of component.
 *
 * @param {Object} state - State of questionnaire.
 * @returns {Object} html - Virtual DOM tree of component.
 */
Results.prototype.render = function(state) {

  var that = this;

  // Get translator service
  var t = this.questionnaire.get('translator');

  var rightAnswers = 0;

  this.questionnaire.questions.forEach(function(question, index) {
    if (question.state.rightAnswer) {
      rightAnswers++;
    }
  });

   var content;
  if ((rightAnswers / this.questionnaire.questions.length) < 0.49){
    content =
          h('div.row', 
            h('div.col-sm-12.text-center', [
              h('h2', t('Harmless Fairy')),
              t('I don’t want to lie, we have seen better. '),
              t(' You answered ') +
                rightAnswers +
              t(' questions of total ') +
                this.questionnaire.questions.length +
                 ' right!',
              t(' But who could be angry at that face. Good news: there is plenty of '), 
              h('a', {
                href: 'https://camunda.com/bpmn/training/'
              }, t('training material')),
              h('br'),
              h('a.twitter-share-button', {
               href: 'https://twitter.com/share',
                attributes: {
                  'data-size':'large', 
                  'data-text':'', 
                  'data-via':'camundaBPM',
                  'data-hashtags':'conqueringBPMN',
                  'data-related':'camundaBPM,bpmn_io',
                  'data-lang':'en' }
              }, t('tweet')),
              h('script', {
                async: true,
                src: 'https://platform.twitter.com/widgets.js',
                charset: 'utf-8'
                }),
              h('img.img-responsive', { src: 'http://placekitten.com/g/1025/600',
                style: {
                  marginTop:'10'
                }
              })
                
          ])
        ); 
   } else if  ((rightAnswers / this.questionnaire.questions.length) < 0.57) {
    content =
          h('div.row', 
            h('div.col-sm-12.text-center', [
              h('h2', t('Trooper')),
              t('Listen up Trooper! '),
              t(' You answered ') +
                rightAnswers +
              t(' questions of total ') +
                this.questionnaire.questions.length +
                 ' right!',
              t(' Put the sandwich down, buckle your seatbelt and start brushing up on your BPMN 2.0 knowledge. To conquer the world of BPMN you need to '),
              h('a', {
                href: 'https://camunda.com/bpmn/training/'
              }, t('train.')),
              h('br'),
              h('a.twitter-share-button', {
               href: 'https://twitter.com/share',
                attributes: {
                  'data-size':'large', 
                  'data-text':'', 
                  'data-via':'camundaBPM',
                  'data-hashtags':'conqueringBPMN',
                  'data-related':'camundaBPM,bpmn_io',
                  'data-lang':'en' }
              }, t('tweet')),
              h('script', {
                async: true,
                src: 'https://platform.twitter.com/widgets.js',
                charset: 'utf-8'
                }),
              h('img.img-responsive', { src: 'http://placekitten.com/g/1025/600',
                style: {
                  marginTop:'10'
                }
            })
                
          ])
        );
  } else if  ((rightAnswers / this.questionnaire.questions.length) < 0.71) {
    content =
          h('div.row', 
            h('div.col-sm-12.text-center', [
              h('h2', t('Champion')),
              t('You are the Champion, you are the Champion... '),
              t(' You answered ') +
                rightAnswers +
              t(' questions of total ') +
                this.questionnaire.questions.length +
                 ' right!',
              t(' No time for losers, because you know your BPMN 2.0 pretty damn well.'),
              h('br'),
              h('a.twitter-share-button', {
               href: 'https://twitter.com/share',
                attributes: {
                  'data-size':'large', 
                  'data-text':'', 
                  'data-via':'camundaBPM',
                  'data-hashtags':'conqueringBPMN',
                  'data-related':'camundaBPM,bpmn_io',
                  'data-lang':'en' }
              }, t('tweet')),
              h('script', {
                async: true,
                src: 'https://platform.twitter.com/widgets.js',
                charset: 'utf-8'
                }),
              h('img.img-responsive', { src: 'http://placekitten.com/g/1025/600',
                style: {
                  marginTop:'10'
                }
            })
                
          ])
        ); 
  }  else if  ((rightAnswers / this.questionnaire.questions.length) < 0.86) {
    content =
          h('div.row', 
            h('div.col-sm-12.text-center', [
              h('h2', t('Hero')),
              t('Come to my savior my hero! '),
              t(' You answered ') +
                rightAnswers +
              t(' questions of total ') +
                this.questionnaire.questions.length +
                 ' right!',
              t(' You managed to master the adventures of BPMN 2.0, while saving a princess, as if it was just a piece of cake!'),
              h('br'),
              h('a.twitter-share-button', {
               href: 'https://twitter.com/share',
                attributes: {
                  'data-size':'large', 
                  'data-text':'', 
                  'data-via':'camundaBPM',
                  'data-hashtags':'conqueringBPMN',
                  'data-related':'camundaBPM,bpmn_io',
                  'data-lang':'en' }
              }, t('tweet')),
              h('script', {
                async: true,
                src: 'https://platform.twitter.com/widgets.js',
                charset: 'utf-8'
                }),
              h('img.img-responsive', { src: 'http://placekitten.com/g/1025/600',
                style: {
                  marginTop:'10'
                }
            })
                
          ])
        );
  } else {
    content =
          h('div.row', 
            h('div.col-sm-12.text-center', [
              h('h2', t('Legendary Conqueror')),
              t('Holy Gateway! You did it! '),
              t(' You answered ') +
                rightAnswers +
              t(' questions of total ') +
                this.questionnaire.questions.length +
                 ' right!',
              t(' Not many people are conquerors, even fewer are legends, you are both! BPMN 2.0 doesn’t control you, you control it.'),
              h('br'),
              h('a.twitter-share-button', {
               href: 'https://twitter.com/share',
                attributes: {
                  'data-size':'large', 
                  'data-text':'', 
                  'data-via':'camundaBPM',
                  'data-hashtags':'conqueringBPMN',
                  'data-related':'camundaBPM,bpmn_io',
                  'data-lang':'en' }
              }, t('tweet')),
              h('script', {
                async: true,
                src: 'https://platform.twitter.com/widgets.js',
                charset: 'utf-8'
                }),
              h('img.img-responsive', { src: 'http://placekitten.com/g/1025/600',
                style: {
                  marginTop:'10'
                }
            })
                
          ])
        ); 

  }
  // Rendering
  var html =
    h('div.bpmn-questionnaire-results.row',
      h('div.col-md-12', [
        h('h2', {style: {'margin-bottom': '20px'}}, 'Ergebnis'),
        content
      ])
    );

  return html;
};

module.exports = Results;
